tests/greenio_test.py

```diff
@@ -634,7 +634,10 @@ def test_skip_nonblocking(self):
         sock1 = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         fd = sock1.fd.fileno()
         flags = fcntl.fcntl(fd, fcntl.F_GETFL)
-        flags = fcntl.fcntl(fd, fcntl.F_SETFL, flags & ~os.O_NONBLOCK)
+        # on SPARC, nonblocking mode sets O_NDELAY as well
+        fcntl.fcntl(fd, fcntl.F_SETFL, flags & ~(os.O_NONBLOCK
+                                                 | os.O_NDELAY))
+        flags = fcntl.fcntl(fd, fcntl.F_GETFL)
         assert flags & os.O_NONBLOCK == 0
```

Comment:
I'm wondering if maybe we should `assert flags & (os.O_NONBLOCK | os.O_NDELAY) == 0` here as well?

Reply:
I suppose that makes sense.